capstone_docs/Maelabord.rmd

```diff
@@ -0,0 +1,753 @@
+                method = "lm", size = 0.5, se = FALSE) +
+    # Stillir táknskýringu fyrir liti og línutegundir
+    scale_color_manual(values = c("Gagnrýnendur " = "magenta", "Áhorfendur " = "seagreen1")) +
+    scale_linetype_manual(values = c("Trendline" = "dashed")) +
+    labs(
+      title = "Kate Winslet: Einkunnir gagnrýnenda vs áhorfenda",
+      x = "Útgáfudagur",
+      y = "Einkunn",
+      color = "",
+      linetype = ""  # Breytir linetype í "Skýring"
+    ) +
+    theme_minimal(base_size = 12) +
+    theme(plot.title = element_text(hjust = 0.5, face = "bold"))
+  
+  ggplotly(trend_comparison_plot_winslet, tooltip = "text")
+})
+
+  
+  #RUNTIME
+  # Audience Rating vs Runtime plot
+output$rating_runtime_plot <- renderPlotly({
+  plot_data <- rotten_tomatoes_leo_kate %>%
+    mutate(
+      both_actors = grepl("Leonardo DiCaprio", actors) & grepl("Kate Winslet", actors),
+      actor = case_when(
+        both_actors ~ "Both",
+        grepl("Leonardo DiCaprio", actors) ~ "Leonardo DiCaprio",
+        grepl("Kate Winslet", actors) ~ "Kate Winslet",
+        TRUE ~ NA_character_
+      )
+    ) %>%
+    filter(!is.na(actor))
+  
+  plot_data <- plot_data %>%
+    mutate(
+      hover_text = paste(
+        "<b>Title:</b>", movie_title,
+        "<br><b>Lengd:</b>", runtime, " mín", 
+        "<br><b>Áhorfendaeinkunn:</b>", audience_rating,
+        "<br><br><b>Info:</b>", str_wrap(movie_info, width = 50) %>% str_replace_all("\n", "<br>")
+      )
+    )
+  
+  p <- ggplot(plot_data, aes(x = audience_rating, y = runtime)) +
+    geom_point(
+      aes(color = actor, shape = actor, text = hover_text),
+      size = 2
+    ) +
+    scale_color_manual(
+      values = c("Leonardo DiCaprio" = "skyblue", 
+                 "Kate Winslet" = "pink", 
+                 "Both" = "gold")
+    ) +
+    scale_shape_manual(
+      values = c("Leonardo DiCaprio" = 16, "Kate Winslet" = 16, "Both" = 8)
+    ) +
+    geom_smooth(
+      aes(group = 1, linetype = "Overall Trend"),  # Skýrari lýsing
+      method = "lm", 
+      color = "grey", 
+      size = 0.5,
+      se = FALSE,
+      show.legend = TRUE  # Virkja skýringu fyrir línutegund
+    ) +
+    scale_linetype_manual(
+      values = c("Overall Trend" = "dashed"), 
+      labels = c("Overall Trend" = "Trendline (All Movies)")
+    ) +
+    labs(
+      title = "Áhrif lengdar á mynd",
+      x = "Áhorfendaeinkunn",
+      y = "Lengd myndar í mínútum",
+      color = "Leikari",
+      shape = "Leikari",
+      linetype = NULL  # Fjarlægir `linetype` sem titil í skýringum
+    ) +
+    guides(
+      linetype = guide_legend(order = 2)  # Stýrir röð trendlínu í skýringunni
+    ) +
+    theme_minimal(base_size = 12) +
+    theme(
+      plot.title = element_text(hjust = 0.5, face = "bold")
+    )
+  
+  # Breytum ggplot í plotly
+  ggplotly(p, tooltip = "text") %>%
+    layout(hoverlabel = list(align = "left")) 
+})

```

Comment:
1. mutate og gagnaflokkun:
   - Mikil endurtekning: Í fyrsta skrefi er verið að flokka gögn fyrir bæði leikara (eða báða í einni mynd). Þetta er mjög algengt og gæti verið skilvirkara ef unnið væri SQL-megin.
   - Tillaga: Bæta við view í gagnagrunninum sem framkvæmir þessar flokkanir fyrir leikarana (Leonardo DiCaprio, Kate Winslet, Both) og einfaldar vinnsluna í R.
   - Dæmi: Í stað þess að nota grepl í hverri skýrslu, búðu til SQL-dálk eða view sem inniheldur flokkaða leikarana.
2. hover_text í R frekar en SQL:
   - Textinn sem er búinn til fyrir hover_text gæti verið unninn í gagnagrunninum. Þetta myndi minnka vinnslu í R og forðast flókna textameðhöndlun í lokaskrefum.
   - Tillaga: Útbúa view í gagnagrunninum sem hefur forhannaðan texta fyrir hover_text, með dálka fyrir viðeigandi upplýsingar (t.d., titill, lengd, einkunn).
3. geom_smooth með línutegund og skýringar:
   - Trendlínan (geom_smooth) er skilmerkilega útfærð, en skýringin (legend) gæti verið bætt.
   - Tillaga: Skýrslutextinn „Trendline (All Movies)“ gæti verið skilvirkari ef hann væri útskýrt betur í mælaborðinu sjálfu, ekki aðeins í línuteikningunni.
4. Hönnun og skilvirkni:
    - Visual Overlap: Þegar gögnin innihalda margar myndir frá báðum leikurum (eða báðum), þá gæti verið að gullpunktar trufli skýrleika.
    - Tillaga: Nota facet_wrap til að aðskilja myndir eftir leikaraflokki (e.g., Both, Leonardo DiCaprio, Kate Winslet). Þetta gæti bætt notendaupplifunina.
5. guides og skýringar:
   - Röðun skýringar fyrir trendlínur (linetype) og punkta er skýrlega útfærð, en til að bæta læsileika væri hægt að lýsa í skýringunni hvers vegna trendlínan er „Overall Trendline“ (e.g., með hönnunartilfelli).
6. Gögn óbreytt fyrir „Audience Rating vs Runtime“:
    - Ef gögnin breytast sjaldan, væri betra að geyma útreikningana fyrirfram og vinna með þau sem uppfærðan gagnagrunn eða gagnatöflu frekar en að framkvæma þessar skipanir í hverri vinnslu.
    - Þetta getur sparað útreikning í real-time.